### ISSUE TITLE
ansible-galaxy install don't fetch properly from git and "--force" param don't work

### ISSUE THREAD

**Issue by alex-left** [ansible/ansible#24441]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

ansible-galaxy
##### ANSIBLE VERSION

```
2.3
```

##### OS / ENVIRONMENT

n/a
##### SUMMARY

i have detect two problems with behavior of "ansible-galaxy install"
first problem: don't fetch properly role from a git repo
second problem: the parameter "--force" don't force the fetching of the role

##### STEPS TO REPRODUCE

``` ansible-galaxy install -r requirements.yml -p roles --force

<!--- Paste example playbooks or commands between quotes below -->
In requirements file: 
```yaml

- src: git+git@bitbucket.org:example/some-repo.git
  name: myrole
  version: v1.0.8

```

##### EXPECTED RESULTS

ansible must fetch the correct tag of the repository and if "--force" is enable, should overwrite existent files. 

##### ACTUAL RESULTS

1. the version of the repo cloned is incorrect
2. if role exists, "--force" param don't overwrite the files:

```- myrole (v1.0.8) is already installed, skipping.
```

**Comment by mensler** [comment]
Fix is tagged as backport in #23391, so hopefully it will make it in one of the 2.3 bugfix releases.

**Comment by gundalow** [comment]
resolved_by_pr 23391

Ansible 2.3.1 RC1 should contain this fix

**Comment by samdoran** [comment]
Doesn't look like this fix made it into 2.3.1.0

**Comment by gundalow** [comment]
Should be in `2.3.2` according to https://github.com/ansible/ansible/pull/23391
